### PR TITLE
Expose intializeRuntime timeout

### PR DESCRIPTION
**Pull request by Thomas101**
Expose timeout option to initializeRuntime as per discussion in https://github.com/kwonoj/hunspell-asm/issues/145#issuecomment-429440681

### CODE REVIEW

File: src/getModuleLoader.ts [hunk 1]
```diff
@@ -30,7 +30,8 @@ type runtimeModuleType = (moduleObject: StringMap) => AsmRuntimeType;
 type getModuleLoaderType = <T, R extends AsmRuntimeType>(
   factoryLoader: (runtime: R, environment: ENVIRONMENT) => T,
   runtimeModule: runtimeModuleType,
-  module?: StringMap
+  module?: StringMap,
+  initializeOpts?: StringMap
```
Comment (kwonoj) [inline src/getModuleLoader.ts]: let's make this to be strictly typed:

```
interface ModuleInitOption {
  timeout: number
};

getModuleLoaderType(
... module?: StringMap,
initOptions: Partial<ModuleInitOption> = {})
```

File: src/getModuleLoader.ts [hunk 2]
```diff
@@ -44,12 +45,15 @@ type getModuleLoaderType = <T, R extends AsmRuntimeType>(
...
+ * @param {{[key: string]: any}} [initializeOpts] Stringmap object to be used to configure the intialization of the module.
+ *
  * @returns {moduleLoaderType<T>} Loader function
  */
 const getModuleLoader: getModuleLoaderType = <T, R extends AsmRuntimeType>(
   factoryLoader: (runtime: R, environment: ENVIRONMENT) => T,
   runtimeModule: runtimeModuleType,
-  module?: StringMap
+  module?: StringMap,
+  initializeOpts?: StringMap
```
Comment (kwonoj) [inline src/getModuleLoader.ts]: and we can destruct directrly:

```
(... {timeout}: Partial<ModuleInitOption>)
```

Comment (Thomas101) [reply src/getModuleLoader.ts]: You'll have to excuse my absolute naivety on this, my typescript-foo is weak. Not too sure what I'm missing here...

```diff
   module?: StringMap,
-  initializeOpts?: StringMap
+  initOptions: (... {timeout}: Partial<ModuleInitOption>)
 ) => async (environment?: ENVIRONMENT) => {
```
```
src/getModuleLoader.ts(64,1): error TS1005: '=>' expected.
```

Comment (kwonoj) [reply src/getModuleLoader.ts]: `...` was just to skip rest of params, so what I meant was

```ts
type getModuleLoaderType = <T, R extends AsmRuntimeType>(
  factoryLoader: (runtime: R, environment: ENVIRONMENT) => T,
  runtimeModule: runtimeModuleType,
  module?: StringMap,
  {timeout}: Partial<InitOptions> = {}
) => moduleLoaderType<T>;
```

File: src/getModuleLoader.ts (hunk 1 above)
Comment (kwonoj) [reply src/getModuleLoader.ts]: and here should be

```
type getModuleLoaderType = <T, R extends AsmRuntimeType>(
  factoryLoader: (runtime: R) => T,
  runtimeModule: runtimeModuleType,
  module?: StringMap,
  {timeout}?: Partial<ModuleInitOption>
) => moduleLoaderType<T>;
```

File: src/getModuleLoader.ts (hunk 2 above)
Comment (Thomas101) [reply src/getModuleLoader.ts]: Thank you :-)

**Pull request merged**